### PR TITLE
Several fixes for the new create dialog

### LINKED ISSUE #2954: Enter keypress defaulting to 'delete my account'
*With extra mouse movement for drama*

term

So, while changing my CartoDB password today I pressed enter to apply the changes, which instead of triggering the 'Save changes' button, it did so with 'Delete my account', which is probably not what we want.

@xavijam

### PR DESCRIPTION
**Pull request by xavijam**
screen shot 2015-03-31 at 09 34 01

screen shot 2015-03-31 at 09 34 12

screen shot 2015-03-31 at 09 35 00

Several tasks fixing problems in new create dialog:

- [x] Review disabled import panes (fixes https://github.com/CartoDB/cartodb/issues/2910).
- [x] Import panes with url input should have a submit button (fixes https://github.com/CartoDB/cartodb/issues/2944).
- [x] HTMLfy map templates layout changes (fixes https://github.com/CartoDB/cartodb/issues/2945).
- [x] Add no-datasets block within dialog when data library is selected and user doesn't have any dataset imported (fixes https://github.com/CartoDB/cartodb/issues/2895).
- [x] Add onboarding layouts for create a map/dataset (fixes https://github.com/CartoDB/cartodb/issues/2883).
- [x] Change mailchimp import texts (fixes https://github.com/CartoDB/cartodb/issues/2895).

And also:

- [x] Enter keypress defaulting to 'delete my account' (Fixes #2954).

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2629/))

**Comment by xavijam** [comment]
@saleiva, change mailchimp texts as you want ;).

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2638/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2643/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2644/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2652/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2663/))

**Comment by xavijam** [comment]
Could you take a look @viddo? Templates changes mainly.

**Comment by xavijam** [comment]
Passing it to @javierarce

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2669/))

File: lib/assets/javascripts/cartodb/new_common/dialogs/create/listing/imports/service_import/import_service_view.js [hunk 1]
```diff
@@ -36,7 +36,8 @@ module.exports = ImportDefaultView.extend({
       ext: false,                 // If files should show extension
       title: 'filename',          // Title attribute
       description: '<%= size %>', // Description attribute
-      formatDescription: 'size'   // If any format function should be applied over the description
+      formatDescription: 'size',  // If any format function should be applied over the description
+      headerTemplate: '',         // Header template
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create/listing/imports/service_import/import_service_view.js]: There's an extra comma here.

File: lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <h3 class="ImportPanel-headerTitle">
+    <% if (state === 'selected') { %>
+      MailChimp campaign selected
+    <% } else { %>
+      Connect with MailChimp
+    <% } %>
+  </h3>
+  <p class="ImportPanel-headerDescription <% if (state === "error") { %>ImportPanel-headerDescription--negative<% } %>">
+    <% if (state === "idle") { %>
+      Log in with your account and select any of your campaigns
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs]: I would add a dot here, like the rest of the messages.

File: lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      Checking Token.
+    <% } %>
+    <% if (state === "oauth") { %>
+      Requesting oAuth.
+    <% } %>
+    <% if (state === "retrieving") { %>
+      A list of your MailChimp campaigns will be displayed in a moment.
+    <% } %>
+    <% if (state === "selected") { %>
+      Campaign selected
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs]: Ditto (see previous message)

File: app/assets/stylesheets/new_common/create/create_templates.css.scss [hunk 4]
```diff
@@ -3,20 +3,60 @@
...
+  position: relative;
+  display: inline-block;
+  font-size: $sFontSize-larger;
+  color: #9B9B9B;
+  background: $cStructure-grayBkg;
+  padding: 0 10px;
+  z-index: 2;
+}
+.CreateDialog-templatesItems { width: $w }
 .CreateDialog-templatesItems { width: 700px }
```
Comment (javierarce) [inline app/assets/stylesheets/new_common/create/create_templates.css.scss]: Not sure if here makes sense using `$w` too.

**Comment by javierarce** [comment]
:+1: code looks legit to me.

**Comment by xavijam** [comment]
Thanks @javierarce, reviewing your comments :)

File: app/assets/stylesheets/new_common/create/create_templates.css.scss (hunk 4 above)
Comment (xavijam) [inline app/assets/stylesheets/new_common/create/create_templates.css.scss]: Oh, it was repeated, good catch. But I use $w for not repeating 700 in several places.

File: lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/views/create/listing/import_types/data_header_mailchimp.jst.ejs]: :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2672/))

**Pull request merged**